tests/flow/union/issue-1249.js

```diff
@@ -0,0 +1 @@
+type Bar = ( number | string ) => number;
```

Comment:
Probably want to move this outside of the flow folder, as these tests are directly imported from flow and will be sync'd in the future.

Maybe as a fixture under `flow_parens`?

(We can also move `flow_return_arrow/parens` as a fixture into that)?